### PR TITLE
Build with CentOS6

### PR DESCRIPTION
**Pull request by imphil**
Build all toolchains on CentOS6 to link against the oldest glibc users of this toolchain might have installed.

The sources for the Docker image used to build the toolchains can be found at https://github.com/lowRISC/lowrisc-it/tree/master/docker/lowrisc-base-centos6.

To enable building I had to disable Qemu builds, which were unused anyways. I do have a partial fix for the Qemu builds around, I'll push that as starting point once we're done here.

Rasmus has confirmed that a toolchain built with this config works on their RHEL6 setup.

### CODE REVIEW

File: build-clang-with-args.sh [hunk 1]
```diff
@@ -50,8 +50,11 @@ cd "${build_top_dir}/build"
 
 
 llvm_dir="${build_top_dir}/build/llvm-project"
-git clone https://github.com/llvm/llvm-project.git "${llvm_dir}"
+if [ ! -d "${llvm_dir}" ]; then
+  git clone https://github.com/llvm/llvm-project.git "${llvm_dir}"
+fi
 cd "${llvm_dir}"
+git fetch origin
 git checkout --force "${LLVM_VERSION}"
```
Comment (luismarques) [inline build-clang-with-args.sh:58]: It can be a separate patch, but perhaps we could take the opportunity to do a shallow clone?

```
git init
git remote add origin https://github.com/llvm/llvm-project.git
git fetch origin <commit>
git reset --hard FETCH_HEAD
```

File: check-tarball.sh [hunk 2]
```diff
@@ -12,7 +12,6 @@
 # was an issue we ran into.
 
 set -e
-set -x
```
Comment (luismarques) [inline check-tarball.sh:15]: Mmmm.....

File: _upload-artifacts.yml [hunk 3]
```diff
@@ -6,12 +6,7 @@ steps:
 # This comes after uploading azure artifacts so we can debug the azure artifact
 # if this step fails.
 - bash: |
-    set -e
-    find "${ARTIFACT_STAGING_DIR}" \
-      -type f \
-      -name "*.tar.xz" \
-      -print0 \
-      | xargs -0 -n1 ./check-tarball.sh
+    for f in ${ARTIFACT_STAGING_DIR}/*.tar.xz; do ./check-tarball.sh "$f"; done
```
[the remark this answers is not in the document]
Comment (asb) [reply _upload-artifacts.yml:9]: The new for loop would have problems if `ARTIFACT_STAGING_DIR` had whitespace in it. Although that directory name is in our control, it would probably be better if it was `for f in "${ARTIFACT_STAGING_DIR}/*.tar.xz";` just to avoid readers having to thing about whether the lack of quoting may be a problem.

Unless I'm missing something, the check-tarball.sh return code is no longer checked is it? (at least, not other than the last invocation of check-tarball.sh)

File: azure-pipelines.yml [hunk 4]
```diff
@@ -18,11 +18,17 @@ jobs:
...
-    vmImage: "ubuntu-16.04"
+    vmImage: "ubuntu-20.04"
+  container: lowrisc/lowrisc-base-centos6:latest
   timeoutInMinutes: 360
   steps:
   - template: "_build-deps.yml"
 
+  - bash: |
+      sudo mkdir -p /tools/riscv
+      sudo chmod 0777 /tools/riscv
```
[the remark this answers is not in the document]
Comment (imphil) [reply azure-pipelines.yml:29]: This directory is only used during building and never makes it into the build artifact, it doesn't matter.

File: check-tarball.sh (hunk 2 above)
Comment (imphil) [reply check-tarball.sh:15]: The script already outputs some diagnostic information, going beyond that was too verbose for me to see what is going on.

File: _upload-artifacts.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (imphil) [reply _upload-artifacts.yml:9]: I couldn't find a good reason for the `find` (and it clearly works without it, as CI shows). Let me do a follow-up to add the quotes (I thought I had that and then saw globbing broken, but that doesn't make sense, so let's try again.)

File: build-clang-with-args.sh (hunk 1 above)
Comment (imphil) [reply build-clang-with-args.sh:58]: Yeah, I thought about it and then went for the minimal solution. I'd prefer a short script/shell function to clone a repository in this way that we can then reuse in the multiple scripts where we need something similar.
